projects/power_selection/src/resistance_to_temp.c

```diff
@@ -0,0 +1,50 @@
+#include "stdint-gcc.h"
+
+// from
+// http://2avrmz2nom8p47cc28p2743e-wpengine.netdna-ssl.com/wp-content/uploads/2010/11/Thermistor_10K-2.pdf
+const double resistance_temperature_in_C[114][2] = {
+  { -39.44, 323839 }, { -38.33, 300974 }, { -37.22, 279880 }, { -36.11, 260410 },
+  { -35.00, 242427 }, { -33.89, 225809 }, { -32.78, 210443 }, { -31.67, 196227 },
+  { -30.56, 183068 }, { -29.44, 170775 }, { -28.33, 159488 }, { -27.22, 149024 },
+  { -26.11, 139316 }, { -25.00, 130306 }, { -23.89, 121939 }, { -22.78, 114165 },
+  { -21.67, 106939 }, { -20.56, 100218 }, { -19.44, 93909 },  { -18.33, 88090 },
+  { -17.22, 82670 },  { -16.11, 77620 },  { -15.00, 72911 },  { -13.89, 68518 },
+  { -12.78, 64419 },  { -11.67, 60592 },  { -10.56, 57017 },  { -9.44, 53647 },
+  { -8.33, 50526 },   { -7.22, 47606 },   { -6.11, 44874 },   { -5.00, 42317 },
+  { -3.89, 39921 },   { -2.78, 37676 },   { -1.67, 35573 },   { -0.56, 33599 },
+  { 0.56, 31732 },    { 1.67, 29996 },    { 2.78, 28365 },    { 3.89, 26834 },
+  { 5.00, 25395 },    { 6.11, 24042 },    { 7.22, 22770 },    { 8.33, 21573 },
+  { 9.44, 20446 },    { 10.56, 19376 },   { 11.67, 18378 },   { 12.78, 17437 },
+  { 13.89, 16550 },   { 15.00, 15714 },   { 16.11, 14925 },   { 17.22, 14180 },
+  { 18.33, 13478 },   { 19.44, 12814 },   { 20.56, 12182 },   { 21.67, 11590 },
+  { 22.78, 11030 },   { 23.89, 10501 },   { 25.00, 10000 },   { 26.11, 9526 },
+  { 27.22, 9078 },    { 28.33, 8653 },    { 29.44, 8251 },    { 30.56, 7866 },
+  { 31.67, 7505 },    { 32.78, 7163 },    { 33.89, 6838 },    { 35.00, 6530 },
+  { 36.11, 6238 },    { 37.22, 5960 },    { 38.33, 5697 },    { 39.44, 5447 },
+  { 40.56, 5207 },    { 41.67, 4981 },    { 42.78, 4766 },    { 43.89, 4561 },
+  { 45.00, 4367 },    { 46.11, 4182 },    { 47.22, 4006 },    { 48.33, 3838 },
+  { 49.44, 3679 },    { 50.56, 3525 },    { 51.67, 3380 },    { 52.78, 3242 },
+  { 53.89, 3111 },    { 55.00, 2985 },    { 56.11, 2865 },    { 57.22, 2751 },
+  { 58.33, 2642 },    { 59.44, 2538 },    { 60.56, 2438 },    { 61.67, 2343 },
+  { 62.78, 2252 },    { 63.89, 2165 },    { 65.00, 2082 },    { 66.11, 2003 },
+  { 67.22, 1927 },    { 68.33, 1855 },    { 69.44, 1785 },    { 70.56, 1718 },
+  { 71.67, 1655 },    { 72.78, 1594 },    { 73.89, 1536 },    { 75.00, 1480 },
+  { 76.11, 1427 },    { 77.22, 1375 },    { 78.33, 1326 },    { 79.44, 1279 },
+  { 80.56, 1234 },    { 81.67, 1190 },    { 82.78, 1149 },    { 83.89, 1109 },
+  { 85.00, 1070 },    { 86.11, 1034 }
+};
+
+double resistance_to_temp(double resistance) {
+  // given resistance, use the table to find the temperature.
+  double temp = UINT16_MAX;
+
+  // find the first resistance value that is lower
+  for (int i = 0; i < 114; ++i) {
```

Comment:
Please use #define for 114, I know it seems trivial but it just makes it that much easier to read. Also please put a quick comment defining the structure of the array, i.e. what's the first value at each index, what's the second. It wouldn't hurt to define an enum (local to the `.c` file) for this I think? Would improve clarity but I won't make you.

Reply:
What do you mean by a local enum?

Reply:
I think it would be more clear if you defined an enum `typedef enum { RES_TEMP_TABLE_TEMP = 0, RES_TEMP_TABLE_RES, NUM_RES_TEMP_TABLE_INDICES } ResTempTableIndex;`, then when you're referring to `resistance_temperature_in_C[i][1]` you can say `resistance_temperature_in_C[i][RES_TEMP_TABLE_RES` to indicate you're accessing the resistance in the table. I think this is more clear than just a 1 or a 0

Reply:
ah i see. I'll do that